Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm

```diff
@@ -390,11 +391,48 @@ - (CFURLRef)_cfurl;
     return { String(), false };
 }
 
+Vector<Cookie> NetworkStorageSession::vectorOfCookiesForSession(const URL& firstParty, const SameSiteInfo& sameSiteInfo, const URL& url, std::optional<FrameIdentifier> frameID, std::optional<PageIdentifier> pageID, IncludeHTTPOnlyOrNot includeHTTPOnly, IncludeSecureCookies includeSecureCookies, ApplyTrackingPrevention applyTrackingPrevention, ShouldRelaxThirdPartyCookieBlocking shouldRelaxThirdPartyCookieBlocking, const CookieStoreGetOptions& options) const
+{
+    ASSERT(hasProcessPrivilege(ProcessPrivilege::CanAccessRawCookies) || m_isInMemoryCookieStore);
+
+    BEGIN_BLOCK_OBJC_EXCEPTIONS
+
+    auto cookies = cookiesForURL(firstParty, sameSiteInfo, url, frameID, pageID, applyTrackingPrevention, shouldRelaxThirdPartyCookieBlocking);
+    if (![cookies count])
+        return { };
+
+    Vector<Cookie> cookiesVector;
+    NSString *name = options.name;
+    for (NSHTTPCookie *cookie in cookies.get()) {
+        if (![[cookie name] length])
+            continue;
+        if (!includeHTTPOnly && [cookie isHTTPOnly])
+            continue;
+        if ([cookie isSecure]) {
+            if (includeSecureCookies == IncludeSecureCookies::No)
+                continue;
+        }
+        if (name && ![[cookie name] isEqualToString:name])
```

Comment:
Still wrong:
```
inline String::operator NSString *() const
{
    if (!m_impl)
        return @"";
    return *m_impl;
}
```

so if `options.name` is a null string, `name` will be an *empty* NSString (not `nil`) and you'll return no cookies instead of all of them.